qiskit_experiments/quantum_volume/qv_experiment.py

```diff
@@ -0,0 +1,261 @@
+            additional_trials (int): The amount of trials to add
+        """
+        self._trials += additional_trials
+
+    @property
+    def trials(self):
+        """Return number of trials in the experiment"""
+        return self._trials
+
+    def _get_ideal_data(self, circuits, run_options):
+        """
+        in case the user do not have aer installed - use Terra to calculate the ideal state
+        Args:
+            circuits: the circuits to extract the ideal data from
+        Returns:
+            dict: data object with the circuit's metadata
+                  and the probability for each state in each circuit
+        """
+        if self._simulation_backend:
+            for circuit in circuits:
+                circuit.metadata["is_simulation"] = True
+                circuit.save_probabilities()
+            return execute(circuits, backend=self._simulation_backend, **run_options)
+        else:
+            from qiskit.quantum_info import Statevector
+            import numpy as np
+
+            sim_obj = []
+            for circuit in circuits:
+                circuit.metadata["is_simulation"] = True
+                state_vector = Statevector(circuit)
+                prob_vector = np.multiply(state_vector, state_vector.conjugate())
+                sim_data = {"probabilities": prob_vector, "metadata": circuit.metadata}
+                sim_obj.append(sim_data)
+            return sim_obj
+
+    # pylint: disable = arguments-differ
+    def circuits(self, backend=None):
+        """Return a list of QV circuits, without the measurement instruction
+        Args:
+            backend (Backend): Optional, a backend object.
+        Returns:
+            List[QuantumCircuit]: A list of :class:`QuantumCircuit`s.
+        """
+        circuits = []
+        depth = self._num_qubits
+
+        # continue the trials numbers from previous experiments runs
+        for trial in range(self._previous_trials + 1, self._trials + 1):
+            qv_circ = QuantumVolume(depth, depth, seed=self._rng)
+            qv_circ.metadata = {
+                "experiment_type": self._type,
+                "depth": depth,
+                "trial": trial,
+                "qubits": self.physical_qubits,
+            }
+            circuits.append(qv_circ)
+
+        return circuits
+
+    def transpiled_circuits(self, backend=None, **kwargs):
+        """Return a list of experiment circuits, before and after transpilation.
+
+        Args:
+            backend (Backend): Optional, a backend object to use as the
+                               argument for the :func:`qiskit.transpile`
+                               function.
+            kwargs: kwarg options for the :meth:`circuits` method, and
+                    :func:`qiskit.transpile` function.
+
+        Returns:
+            List[QuantumCircuit]: A list of :class:`QuantumCircuit`s after transpile.
+            List[QuantumCircuit]: A list of :class:`QuantumCircuit`s before transpile.
```

Comment:
You need both  `transpiled_circuits` before transpilation and `circuits` since the first in for Aer ideal simulator and the second in for Terra's `Statevector` class? Please add a comment explaining this.
Also, later add a test that verifies that both options provide the same output.

Reply:
I added a comment.
the transpiled_circuits are used for the real backend, not for a simulator. the circuits without transpilation are for simulator (or Terra's StateVector class, depends whether the user have Aer installed).

therefore - I don't agree that we need to add a test which checks that circuit before and after transpilation are the same, because the circuits doesn't have to be the same, and their results doesn't have to be the same - it is up to the transpiler the user chooses to use (and the user will probably use the highest optimization level possible)

Reply:
I meant a test that the ideal simulation of the **original** circuits using Aer and using the Statevector class give the same output